src/hotspot/share/runtime/frame.cpp

```diff
@@ -1163,8 +1163,7 @@ void frame::oops_do_internal(OopClosure* f, NMethodClosure* cf,
 #ifndef PRODUCT
   // simulate GC crash here to dump java thread in error report
   if (CrashGCForDumpingJavaThread) {
-    char *t = nullptr;
-    *t = 'c';
+    guarantee(!CrashGCForDumpingJavaThread, "");
```

Comment:
The surrounding `if` is no longer needed.

Reply:
Thanks, the curly braces has been removed.